tool/plugin/lib/plugin.dart

```diff
@@ -210,9 +250,62 @@ class DeployCommand extends ProductCommand {
   String get description => 'Upload the Flutter plugin to the JetBrains site.';
 
   Future<int> doit() async {
+    if (isReleaseMode) {
+      if (!performReleaseChecks()) {
+        return new Future(() => 1);
+      }
+    } else {
+      log('Deploy must have a --release argument');
+      return new Future(() => 1);
+    }
+    String password;
+    try {
+      // Detect test mode early to keep stdio clean for the test results parser.
+      bool mode = stdin.echoMode;
+      stdout.writeln(
+          'Please enter the username and password for the JetBrains plugin repository');
+      stdout.write('Username: ');
+      username = stdin.readLineSync();
+      stdout.write('Password: ');
+      stdin.echoMode = false;
+      password = stdin.readLineSync();
+      stdin.echoMode = mode;
+    } on StdinException {
+      password = "hello"; // For testing.
+      username = "test";
+    }
 
-    // TODO(messick): implement
-    throw 'unimplemented';
+    Directory directory = Directory.systemTemp.createTempSync('plugin');
+    tempDir = directory.path;
+    var file = new File('$tempDir/.content');
+    file.writeAsStringSync(password, flush: true);
+
+    var value = 0;
+    for (var spec in specs) {
+      String filePath = archiveFilePath(spec);
+      value = await upload(filePath, plugins[spec.pluginId]);
+      if (value != 0) return value;
```

Comment:
Will `upload()` have already printed a failure message?

Reply:
Possibly. The most likely failure is that the file hasn't been built and it checks for that. There is currently no error printed if `curl` fails but there should be.